test/snapshot-order/backwards-compatibility.js

```diff
@@ -0,0 +1,119 @@
+const test = require('@ava/test');
+const exec = require('../helpers/exec');
+const fs = require('fs');
+const path = require('path');
+const getSnapshotIds = require('./helpers/get-snapshot-ids');
+
+const unsortedSnapshotPath = path.join(__dirname, 'fixtures', 'backwards-compatibility', 'unsorted', 'test.js.snap');
+const unsortedReportPath = path.join(__dirname, 'fixtures', 'backwards-compatibility', 'unsorted', 'test.js.md');
+
+const unsortedSnapshot = fs.readFileSync(unsortedSnapshotPath);
+const unsortedReport = fs.readFileSync(unsortedReportPath);
+
+test('unsorted snapshots are unchanged when checking', async t => {
+	const options = {
+		cwd: exec.cwd('backwards-compatibility', 'check-only'),
+		env: {
+			AVA_FORCE_CI: 'not-ci'
+		}
+	};
+
+	// Schedule snapshot cleanup
+	t.teardown(() => {
+		fs.unlinkSync(snapshotPath);
+		fs.unlinkSync(reportPath);
+	});
+
+	const snapshotPath = path.join(options.cwd, 'test.js.snap');
+	const reportPath = path.join(options.cwd, 'test.js.md');
+
+	// Install a known-unsorted snapshot, report
+	fs.copyFileSync(unsortedSnapshotPath, snapshotPath);
+	fs.copyFileSync(unsortedReportPath, reportPath);
```

Comment:
Write the buffers from memory, instead of copying the files? Makes the comparison below clearer I think.